pkg/scraping/client.go

```diff
@@ -55,3 +56,24 @@ func (c *nodeClient) stateHash(ctx context.Context, height int) (*proto.StateHas
 	}
 	return sh, nil
 }
+
+func (c *nodeClient) baseTarget(ctx context.Context, height int) (int, error) {
+
+	_, resp, err := c.cl.Blocks.HeadersAt(ctx, uint64(height))
+	if err != nil {
+		nodeURL := c.cl.GetOptions().BaseUrl
+		c.zap.Error("headers at request failed", zap.String("node", nodeURL), zap.Error(err))
+		return 0, err
+	}
+	b := struct {
```

Comment:
This temporary structure is not required at all. Decoded block headers contains base target value already. But you put it under `_`. It's in the first result.